pkg/builder/builder.go

```diff
@@ -575,19 +586,24 @@ func makeServerContainer(vdb *vapi.VerticaDB, sc *vapi.Subcluster) corev1.Contai
 	}...)
 
 	if vmeta.UseVClusterOps(vdb.Annotations) {
-		envVars = append(envVars, []corev1.EnvVar{
-			// Old model is to provide the path to each of the certs that are
-			// mounted in the container.
-			{Name: "NMA_ROOTCA_PATH", Value: fmt.Sprintf("%s/%s", paths.HTTPServerCertsRoot, paths.HTTPServerCACrtName)},
-			{Name: "NMA_CERT_PATH", Value: fmt.Sprintf("%s/%s", paths.HTTPServerCertsRoot, corev1.TLSCertKey)},
-			{Name: "NMA_KEY_PATH", Value: fmt.Sprintf("%s/%s", paths.HTTPServerCertsRoot, corev1.TLSPrivateKeyKey)},
-			// New model is for the NMA to read the secrets directly from k8s.
-			// We provide the secret namespace and name for this reason. Once
-			// implemented we no longer need to provide the above environment
-			// variables.
-			{Name: NMASecretNamespaceEnv, Value: vdb.ObjectMeta.Namespace},
-			{Name: NMASecretNameEnv, Value: vdb.Spec.NmaTLSSecret},
-		}...)
+		if vmeta.UseNMACertsMount(vdb.Annotations) {
+			envVars = append(envVars, []corev1.EnvVar{
+				// Old model is to provide the path to each of the certs that are
+				// mounted in the container.
+				{Name: NMARootCAEnv, Value: fmt.Sprintf("%s/%s", paths.NMACertsRoot, paths.HTTPServerCACrtName)},
+				{Name: NMACertEnv, Value: fmt.Sprintf("%s/%s", paths.NMACertsRoot, corev1.TLSCertKey)},
+				{Name: NMAKeyEnv, Value: fmt.Sprintf("%s/%s", paths.NMACertsRoot, corev1.TLSPrivateKeyKey)},
+			}...)
+		} else {
+			envVars = append(envVars, []corev1.EnvVar{
+				// New model is for the NMA to read the secrets directly from k8s.
+				// We provide the secret namespace and name for this reason. Once
+				// implemented we no longer need to provide the above environment
+				// variables.
```

Comment:
We should update the comment.
```suggestion
				// The NMA will read the secrets directly from k8s. We provide the
				// secret namespace and name for this reason.
```